File: SOFTWARE/AutorizadorCanales/AutorizadorCanales.Domain/Entidades/CF/Agencia.cs
namespace AutorizadorCanales.Domain.Entidades.CF;

public class Agencia : EntidadEmpresa
{
    #region Propiedades
    /// <summary>
    /// Código de agencia
    /// </summary>
    public string CodigoAgencia { get; private set; } = null!;
    /// <summary>
    /// Nombre de agencia
    /// </summary>
    public string NombreAgencia { get; private set; } = null!;
    /// <summary>
    /// Indicador de registro activo.
    /// </summary>
    public string? Activo { get; private set; } = null!;
    /// <summary>
    /// Dirección de la agencia.
    /// </summary>
    public string Direccion { get; private set; } = null!;
    /// <summary>
    /// Nombre de la ciudad de la agencia.
    /// </summary>
    public string? NombreCiudad { get; private set; } = null!;
    /// <summary>
    /// Ubigeo de la agencia.
    /// </summary>
    public string CodigoUbigeo { get; private set; } = null!;
    /// <summary>
    /// Indicador del estado de la agencia.
    /// </summary>
    public string Estado { get; private set; } = null!;
    #endregion

    #region Contantes
    public const string PRINCIPAL = "01";
    #endregion
}

File: SOFTWARE/APPMovilCanales/lib/features/emision_giros/models/agencia.dart
class Agencia {
  final String codigoAgencia;
  final String nombreAgencia;
  final String? direccion;

  Agencia({
    required this.codigoAgencia,
    required this.nombreAgencia,
    this.direccion
  });

  factory Agencia.fromJson(Map<String, dynamic> json) => Agencia(
        codigoAgencia: json["CodigoAgencia"],
        nombreAgencia: json["NombreAgencia"],
        direccion: json["Direccion"],
      );

  Map<String, dynamic> toJson() => {
        "CodigoAgencia": codigoAgencia,
        "NombreAgencia": nombreAgencia,
        "Direccion": direccion
      };
}
